### PR TITLE
Add NCI Thesaurus

### PR DESCRIPTION
**Pull request by jsstevenson**
Howdy! We're interested in potentially making use of this in a few of our projects. If you're receptive to PRs, I have a small handful of other sources that we draw from, in addition to NCIt (and let me know if I'm missing anything here).

### CODE REVIEW

File: src/bioversions/sources/ncit.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    bioregistry_id = "ncit"
+    name = "National Cancer Institute Thesaurus"
+    version_type = VersionType.other
+
+    def get(self) -> str:
+        """Get the latest NCIt version number."""
+        soup = get_soup(URL)
+        version = soup.find("input", {"id": "version"}).attrs["value"]
+        return version
```
Comment (cthoyt) [inline src/bioversions/sources/ncit.py:25]: it appears there's also a release date on this page. The text reads `Version:21.08e (Release date:2021-08-30)`. If you parse the date too, you could return them both in a dictionary like in the [DisGeNet getter](https://github.com/biopragmatics/bioversions/blob/main/src/bioversions/sources/disgenet.py)

**Review by cthoyt — CHANGES_REQUESTED** [review]
Everything looks good! I suppose this was a quite simple one :) I added a request to extract the date of the current release as well. 

Last thing - do you know if there is a specific page corresponding to each version? For example in BioGRID, there's a way to construct a URL for a given version, which is really nice. If that's possible it would be great, but not required because we all know NCIt is very difficult to figure out.

File: src/bioversions/sources/ncit.py (hunk 1 above)
Comment (jsstevenson) [reply src/bioversions/sources/ncit.py]: Done!

**Comment by jsstevenson** [comment]
Unfortunately the NCIt FTP archives follow a folder structure that is a little hard to capture in a single f-string -- they place the current year's releases one level up from prior years (which are all housed in subdirectories for each year), eg

```
2020/20.11e Release/
2020/20.12d Release/
21.08e Release/
```

I'd definitely be interested in getting in touch -- one of our group's broader projects focuses on knowledgebase integration in the cancer variant interpretation space (https://cancervariants.org/projects/integration/), so we have a vested interest in things like data provenance and reproducibility.

**Comment by cthoyt** [comment]
I'm going to merge now but if you could send a link to that FTP address I would appreciate it

**Comment by jsstevenson** [comment]
> I'm going to merge now but if you could send a link to that FTP address I would appreciate it

👍 

https://evs.nci.nih.gov/ftp1/NCI_Thesaurus/archive/

**Pull request merged**